x-pack/libbeat/management/managerV2.go

```diff
@@ -605,59 +621,59 @@ func (cm *BeatV2Manager) reload(units map[unitKey]*client.Unit) {
 	}
 	if err != nil {
 		// Output creation failed, there is no point in going any further
-		// because there is no output read the events.
+		// because there is no output to read events.
 		//
 		// Trying to start inputs will eventually lead them to deadlock
 		// waiting for the output. Log input will deadlock when starting,
 		// effectively blocking this manager.
-		err = fmt.Errorf("could not start output: %w", err)
-		outputUnit.UpdateState(client.UnitStateFailed, err.Error(), nil)
-		cm.status = lbmanagement.Failed
-		cm.message = err.Error()
-
-		// If there are any other errors, set the status accordingly.
-		// If len(errs), then the there were no previous and the only
-		// error has been reported already.
-		if len(errs) > 0 {
-			errs = append(errs, err)
-			cm.message = fmt.Sprintf("%s", errs)
+		cm.logger.Errorw("could not start output", "error", err)
+
+		msg := fmt.Sprintf("could not start output: %s", err)
+		if err := outputUnit.UpdateState(client.UnitStateFailed, msg, nil); err != nil {
+			cm.logger.Errorw("setting output state", "error", err)
 		}
+
 		return
 	}
 
+	if err := outputUnit.UpdateState(client.UnitStateHealthy, "Healthy", nil); err != nil {
+		cm.logger.Errorw("setting output state", "error", err)
+	}
+
 	// compute the input configuration
 	//
 	// in v2 only a single input type will be started per component, so we don't need to
 	// worry about getting multiple re-loaders (we just need the one for the type)
 	if err := cm.reloadInputs(inputUnits); err != nil {
-		errs = append(errs, err)
+		merror := &multierror.MultiError{}
+		if errors.As(err, &merror) {
+			for _, err := range merror.Errors {
+				unitErr := cfgfile.UnitError{}
+				if errors.As(err, &unitErr) {
+					unitErrors[unitErr.UnitID] = append(unitErrors[unitErr.UnitID], unitErr.Err)
+					delete(healthyInputs, unitErr.UnitID)
+				}
+			}
+		}
 	}
 
 	// report the stopping units as stopped
 	for _, unit := range stoppingUnits {
 		_ = unit.UpdateState(client.UnitStateStopped, "Stopped", nil)
 	}
 
-	// any error during reload changes the whole state of the beat to failed
-	if len(errs) > 0 {
-		cm.status = lbmanagement.Failed
-		cm.message = fmt.Sprintf("%s", errs)
-	}
-
 	// now update the statuses of all units
```

Comment:
Not part of this PR but just noticed it during the review.  This should probably say "all inputs" or "all input units".

Reply:
I'll rephrase the whole thing, I believe it can be more clear in general.